docs/quality-assuarance-guide.md

```diff
@@ -0,0 +1,144 @@
+    - The `Clean Code` Book can be considered as a good reference.
+    - Also refer
+      to [Ballerina anti-patterns](https://docs.google.com/document/d/1y6QVqaZzZt9jMpYV4jP5WRS_W_KoC4y40Uuoh1ALu8E/edit?usp=sharing)
+      documentation.
+
+- If the extended library module contains a native (i.e. Java) implementation,
+    - Java coding standards and best practices must be followed.
+    - Java static analysis tools such as [Checkstyle](https://checkstyle.sourceforge.io/)
+      and [SpotBugs](https://spotbugs.github.io/) should be configured to ensure code quality.
+- Appropriate comments and documentation should be included in the code for maintainability.
+- Necessary code reviews and peer reviews should be conducted whenever adding new features or making significant
+  changes to the existing code.
+
+## Functionality
+
+- All the library APIs in the module should work as expected.
+    - Edge cases and error scenarios should be appropriately handled within the functions.
+    - Unit tests should be added to verify the functionality of each API.
+- All required features and functionalities should be implemented.
+    - Better to create a proposal document and get it approved before starting the implementation.
+- Any known issues or limitations with the module should be proactively identified and addressed/documented properly.
+
+## Compatibility
+
+- The module should be tested with different versions of Ballerina to ensure compatibility. For handwritten ballerinax
+  modules, There should be separate branches for each Swan Lake update version so that separate workflows can be
+  configured for each version to ensure the compatibility.
+- The `Package.md` file of the module should include an updated compatibility section with the following information:
+    - minimum-compatible Ballerina version (e.g., Swan Lake 2201.4.1)
+    - version of the external library or API that the module is compatible with. (e.g., Azure Service Bus SDK 7.13.1)
+- The `Module.md` should contain clear instructions on how to set up compatible prerequisites (e.g., accounts, access
+  tokens, etc.) for the module to work.
+
+## Documentation
+
+- Each module should have a `Package.md` file which should contain the following information:
+    - Brief overview of the module.
+    - Compatibility section that contains the minimum-compatible Ballerina version and the version of the external
+      library or API that the module is compatible with.
+    - Community links such as the repository, issue tracker, and discussion channels.
+    - The recommended template for the Package.md file can be found in [here](file-templates/Package.md)
+- Each module should have a `Module.md` file which should contain the following information:
+    - Brief overview of the module.
+    - List of prerequisites (e.g., accounts, access tokens, etc.) required to use the module.
+    - Quick start guide that contains a simple example of how to use the module.
```

Comment:
Only code snippets right? Not a complete working sample?

Reply:
we usually use code snippets of an simple examples to explain it step-by-step right? I'll alter the point to mention it properly

Reply:
Changed with https://github.com/ballerina-platform/ballerina-extended-library/pull/514/commits/f92f7644b6b5382042b36a9f8db861044363258d